storage/core/src/main/java/com/hortonworks/streamline/storage/CacheBackedStorageManager.java

```diff
@@ -52,32 +52,51 @@ public void init(Map<String, Object> properties) {
     @Override
     public void add(Storable storable) throws StorageException {
         writer.add(storable);
-        cache.put(storable.getStorableKey(), storable);
+        if (storable.isCacheable()) {
+            cache.put(storable.getStorableKey(), storable);
+        }
     }
 
     @Override
     public <T extends Storable> T remove(StorableKey key) throws StorageException {
-        writer.remove(key);
-        final T oldVal = (T) cache.get(key);
-        cache.remove(key);
-        return oldVal;
+        Storable storable = (Storable) writer.remove(key);
+        if (storable != null && storable.isCacheable()) {
+            storable = cache.get(key);
```

Comment:
I refactored the existing code which was like below, so I am not sure why was it that way  (may be we assumed the cache has the latest value which could be out of sync with the db?)
```java
writer.remove(key);
final T oldVal = (T) cache.get(key);
cache.remove(key);
return oldVal;
```